packages/aws-cdk-lib/aws-autoscaling/lib/auto-scaling-group.ts

```diff
@@ -1280,51 +1282,71 @@ export class AutoScalingGroup extends AutoScalingGroupBase implements
         throw new Error('Setting \'instanceType\' is required when \'launchTemplate\' and \'mixedInstancesPolicy\' is not set');
       }
 
+      Tags.of(this).add(NAME_TAG, this.node.path);
+
       this.securityGroup = props.securityGroup || new ec2.SecurityGroup(this, 'InstanceSecurityGroup', {
         vpc: props.vpc,
         allowAllOutbound: props.allowAllOutbound !== false,
       });
-      this._connections = new ec2.Connections({ securityGroups: [this.securityGroup] });
-      this.securityGroups = [this.securityGroup];
-      Tags.of(this).add(NAME_TAG, this.node.path);
 
       this._role = props.role || new iam.Role(this, 'InstanceRole', {
         roleName: PhysicalName.GENERATE_IF_NEEDED,
         assumedBy: new iam.ServicePrincipal('ec2.amazonaws.com'),
       });
-
       this.grantPrincipal = this._role;
 
       if (props.ssmSessionPermissions) {
         this.role.addManagedPolicy(iam.ManagedPolicy.fromAwsManagedPolicyName('AmazonSSMManagedInstanceCore'));
       }
 
-      const iamProfile = new iam.CfnInstanceProfile(this, 'InstanceProfile', {
```

Comment:
We need a way to keep using this with the new `LaunchTemplate`

Reply:
I came up with a solution to keep using this with the new `LaunchTemplate` - this will be reflected when all changes are pushed